src/olympia/addons/serializers.py

```diff
@@ -1196,11 +1196,12 @@ def create(self, validated_data):
         # Add categories
         addon.set_categories(validated_data.get('all_categories', []))
         addon.set_tag_list(validated_data.get('tag_list', []))
-
-        self.fields['version'].create(
+        addon.version = self.fields['version'].create(
             {**validated_data.get('version', {}), 'addon': addon}
         )
-
+        # When creating, always return the version we just created in the
+        # representation. It uses <instance>.version.
+        self.fields['version'].write_only = False
```

Comment:
Add a comment next to the field declaration too, where we define it as `write_only=True`?